### PR TITLE
Add Wish List module

### PR DESCRIPTION
**Pull request by teeshter**
Added wish list module as requested in issue #502

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+                var resultModel = new AddToWishListResult();
+
+                var product = await _productRepository
+                    .Query()
+                    .Include(x => x.ThumbnailImage)
+                    .SingleOrDefaultAsync(x => x.Id == model.ProductId);
+
+                if (product == null)
+                {
+                    return new NotFoundResult();
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs]: should be return NotFound(); for consistency and similar for other places

File: src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        {
+            var user = await _workContext.GetCurrentUser();
+
+            var wishList = await _wishListRepository
+                .Query()
+                .SingleOrDefaultAsync(x => x.SharingCode == id);
+
+            if (wishList == null)
+            {
+                return new NotFoundResult();
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs]: should be return NotFound(); for consistency

File: src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                {
+                    var wishListItem = new WishListItem()
+                    {
+                        WishListId = wishList.Id,
+                        ProductId = model.ProductId,
+                        Quantity = model.Quantity
+                    };
+
+                    _wishListItemRepository.Add(wishListItem);
+                    await _wishListItemRepository.SaveChangesAsync();
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs]: We should call SaveChangesAsync() once for entire this method.

File: src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+                .Query()
+                .SingleOrDefaultAsync(x => x.Id == id);
+
+            if (wishList == null || wishListItem == null)
+            {
+                return new NotFoundResult();
+            }
+
+            _wishListItemRepository.Remove(wishListItem);
+            await _wishListItemRepository.SaveChangesAsync();
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs]: Call SaveChangesAnync once

**Comment by teeshter** [comment]
@thiennn 

I can't say I've seen the wish list work for any guest user on any e-commerce site I've visited. 

Can you give me more details on what you'd like the guest to be able to do? Do you want them to only be able to just add to the wish list, but when an account is created they can then edit and share it? Or do you want the guest to be able to do everything?

Thanks.

File: src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs (hunk 1 above)
Comment (teeshter) [reply src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs]: @thiennn 

Do you mean to replace all places in the controller where NotFoundResult() is used with NotFound(), or only in the two places you actually pointed out?

Also, when do you prefer to use NotFoundResult() versus NotFound()? I saw a lot of places in the existing code that are using NotFoundResult().

Thanks.

Comment (thiennn) [reply src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs]: All the places should use NotFound(); I will eventually update to NoFound() whenever I have a change.

**Comment by thiennn** [comment]
@teeshter , I just don't see any reason why we should limit this featured to only login-ed users. Wait, the wishlist is not sync across devices if the users are not login. Ok, your are right, login should be required to use this features. Thanks

**Comment by teeshter** [comment]
@thiennn 

Please see my changes. Thanks.

**Pull request merged**